client/blocks/comment-button/index.jsx

```diff
@@ -39,7 +45,7 @@ const CommentButton = React.createClass( {
 	render() {
 		let label;
 		const containerTag = this.props.tagName,
-			commentCount = this.props.commentCount;
+			commentCount = ( this.props.syncedCount === undefined ? this.props.commentCount : this.props.syncedCount );
```

Comment:
> We are using redux to hold posts state already, it's just not used yet for the main post-cards in /posts (but it is used for the draft items in the sidebar).

You're right, sorry, I overgeneralized :)

As for the two count props, I answered that in [my main comment below](https://github.com/Automattic/wp-calypso/pull/7581#issuecomment-243971060).